lib/kino/counter.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Kino.Counter do
+  @moduledoc false
+
+  @table_name __MODULE__
+
+  @doc """
+  Initializes resources for global counters.
+  """
+  @spec initialize() :: :ok
+  def initialize() do
+    :ets.new(@table_name, [:set, :named_table, :public])
+    :ok
+  end
+
+  @doc """
+  Increments the given counter and returns the new value.
+  """
+  @spec next(term()) :: integer()
+  def next(key) do
+    :ets.update_counter(@table_name, key, {2, 1}, {key, 0})
+  end
+
+  @doc """
+  Bumps the counter to `value` unless it already has a higher value.
+
+  Returns the current counter value.
+  """
+  @spec bump(term(), integer()) :: integer()
+  def bump(key, value) do
```

Comment:
`bump` typically means "bump by value". Maybe set or put would be better here? :)

Reply:
`set` and `put` sound like they would always override the value, while in this case we only do it if it increases :thinking:

Reply:
@jonatanklosko `at_least`, `update_max`, `put_max`, all work. Your call. :)

Reply:
I mean they all work _in addition_ to what you suggested. :)